### ISSUE TITLE
Help needed: `no-unsafe-type-assertion` in type guard

### ISSUE THREAD

**Issue by Samuel-Beslogic** [danielnixon/eslint-plugin-total-functions#592]
Hi.
I don't understand why this snippet of code is flagged:
`export const excludeNullish = (value => value != null) as <T>(value: T | null | undefined) => value is T;`
![image](https://user-images.githubusercontent.com/84519004/170175470-abdb2fbf-36e2-46d0-bd22-6adc499182ef.png)

It is a function that takes any value as an input, and returns a boolean based on if the value was nullish. Then also let's TypeScript know about the asserted type. Quite useful in arrays and rxjs filters `[null, undefined, 0, 'hi'].filter(excludeNullish)`

**Comment by danielnixon** [comment]
Your `excludeNullish` isn't checking for undefined. Try this:
```ts
export const excludeNullish = <T>(value: T | undefined | null): value is T => {
    return value !== null && value !== undefined;
};
```

**Comment by Samuel-Beslogic** [comment]
Except that it does. `==` on a nullish value is a nullish check. So it does check for undefined. 

But with the way you wrote it I just realized how to write this without the cast. Which is cleaner. Thanks. I was having issues with the typeguard syntax.
```js
export const excludeNullish = <T>(value: T | null | undefined): value is T => value != null;
```

**Comment by danielnixon** [comment]
You're right, sorry. I missed the `==`. That's something else I ban with ESLint 🙃. Glad I could help.
